hs_rest_api/resources/file_metadata.py

```diff
@@ -0,0 +1,200 @@
+from django.shortcuts import get_object_or_404
+
+from rest_framework.response import Response
+from rest_framework import generics
+from rest_framework import serializers
+from rest_framework.exceptions import APIException, PermissionDenied, NotFound
+
+from hs_core.models import ResourceFile
+from hs_core.views.utils import authorize, ACTION_TO_AUTHORIZE
+
+
+# TODO: Once we upgrade past Django Rest Framework 3.3, this won't be necessary
+class JSONSerializerField(serializers.Field):
+    """ Serializer for JSONField -- required to make field writable"""
+
+    def to_internal_value(self, data):
+        return data
+
+    def to_representation(self, value):
+        return value
+
+
+class FileMetaDataSerializer(serializers.Serializer):
+    title = serializers.CharField(required=False)
+    keywords = JSONSerializerField(required=False)
+    spatial_coverage = JSONSerializerField(required=False)
+    extra_metadata = JSONSerializerField(required=False)
+    temporal_coverage = JSONSerializerField(required=False)
+
+
+class FileMetaDataRetrieveUpdateDestroy(generics.RetrieveUpdateDestroyAPIView):
+    serializer_class = FileMetaDataSerializer
+    allowed_methods = ('GET', 'PUT')
+
+    def get(self, request, pk, file_id):
+        """
+        Get a resource file's metadata.
+
+        ## Parameters
+        * `id` - alphanumeric uuid of the resource, i.e. cde01b3898c94cdab78a2318330cf795
+        * `file_id` - integer id of the resource file. You can use the `/hsapi/resource/{id}/files`
+        to get these
+
+        ## Returns
+        ```
+        {
+            "keywords": [
+                "keyword1",
+                "keyword2"
+            ],
+            "spatial_coverage": {
+                "units": "Decimal degrees",
+                "east": -84.0465,
+                "north": 49.6791,
+                "name": "12232",
+                "projection": "WGS 84 EPSG:4326"
+            },
+            "extra_metadata": {
+                "extended1": "one"
+            },
+            "temporal_coverage": {
+                "start": "2018-02-22",
+                "end": "2018-02-24"
+            },
+            "title": "File Metadata Title"
+        }
+        ```
+        """
+        resource_file = get_object_or_404(ResourceFile, id=file_id)
```

Comment:
I forgot to mention last time. ```get()``` also needs to check user access permission on the resource.
Ref: https://github.com/hydroshare/hydroshare/blob/develop/hs_core/views/resource_metadata_rest_api.py#L176